src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +58,98 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    boolean hasWanIp = false;
+    final String uninitializedWan = "0.0.0.0";
+
+    if (NetManager.getWanIp().equalsIgnoreCase(uninitializedWan)) {
+      hasWanIp = true;
+    }
+    return hasWanIp;
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be establlshed if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeOutMinutes returns true if an initialized WAN IP is detected
+   */
+  public static boolean waitForWanIp(int timeOutMinutes) {
+
+    // There are 20 seconds in a wait period. 3 periods exist in a minute
+    int minutesToPeriodsConverter = 3;
+    int maxWaitPeriods = timeOutMinutes * minutesToPeriodsConverter;
+
+    boolean wanConnectionEstablished = false;
+    // Ensure there is a valid WAN IP address
+    for (int i = 0; i < maxWaitPeriods; i++) {
+      Logger.LOG_INFO("WAN IP is: " + NetManager.getWanIp());
+
+      if (deviceHasWanIP()) {
+        // If no connection is established by end of wait time
+        if (i == (maxWaitPeriods - 1)) {
+          Logger.LOG_CRITICAL(
+              "No WAN connection detected, please ensure there is a WAN connection before starting"
+                  + " the connector.");
+          wanConnectionEstablished = false;
+          i = maxWaitPeriods;
+        }
+        // Wait 20 seconds on WAN connection initialization
+        waitForWanHelper();
```

Comment:
```suggestion
        // Wait 20 seconds on WAN connection initialization
        final int waitTimeSeconds = 20;
        waitForWanHelper(waitTimeSeconds);
````

Same as below suggestion